docs/DesignDocs/NotesOnAttestationAPI.md

```diff
@@ -0,0 +1,160 @@
+platform, won’t be able to verify this SGX report.
+- The target info is SGX specific, and is not parsed nor interpreted
+by the TEE-agnostic application code. It is handled by the SGX local
+attestation plugins underneath the `oe_get_evidence()` and
+`oe_verify_evidence()` API in the attester and the verifier respectively.
+
+# Attester – Verifier Security Model and TEE Agnostic Design
+
+- OE SDK is designed for attester and verifier software to be TEE agnostic,
+so that the same source code can be built for and run on different platforms
+- But owners of attesters and verifiers define their security policies,
+and the policies are not TEE agnostic. For example:
+  - The policies define which TEEs, platforms and configurations are trusted
+  to run attesters and verifiers.
+    - They also define security properties of these entities, e.g. keypairs
+    and certs, service provider ID / SPID for SGX EPID evidence, etc.
+    These security properties are required for attestation and secure
+    communication.
+  - They define which evidence formats attesters accept, and configure the
+  list to the verifiers.
+    - Every acceptable evidence format is TEE specific.
+- OE SDK provides means for the attesters and verifiers to be built and
+configured for specified TEEs and platforms, and to generate or verify
+the evidence formats according to their defined security policies.
+  - The OE SDK attestation API enables the attesters and verifiers to discover
+  and register supported plugins that are built, configured, and working
+  properly. For a specific attester or verifier instance, the supported list
+  of evidence is not required to perfectly match the list defined in the
+  security policies.
+
+# Options for Enclave-side Plugin Library Initialization
+
+As described in the design document
+[Attestation: OE SDK Integration with Intel® SGX SDK quote-ex Library for Generation of Evidence in New Formats](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/SGX_QuoteEx_Integration.md),
+enclave-side plugin library initialization of attester plugins can be
+triggered by application enclave call of the OE SDK API
+`oe_get_attester_plugins()`.
+During initialization, the plugin library enumerates supported evidence
+formats, and returns a list of supported attester plugins.
+
+In general, there are several options for initialization of enclave-side
+plugin libraries while keeping application enclaves TEE agnostic.
+
+In this section, we focus on the discussion of the options for initialization
+of attester plugins. Similar options are available for initialization of
+verifier plugins.
+
+1. Every enclave-side plugin library implements the same OE SDK API
+`oe_get_attester_plugins()`.
+
+    With this option, a build of an application enclave project can only link
+    with a single attestation plugin library, to avoid function name collision.
+
+2. Compile-time macro to convert from TEE-agnostic API to TEE-specific
+implementation.
```

Comment:
This option requires the plugin author to define the macro and may not be scalable for unknown plugins.

Reply:
Agreed it has scalability issue.

Reply:
What about adding a phrase about Yen's point, maybe at the end of 226?  Up to you.

Reply:
Thanks! Added a sentence.